.github/workflows/deploy-sandbox.yml

```diff
@@ -37,37 +46,92 @@ jobs:
           ./build-tools/bin/build_tools \
           auth \
           --mfa-arn ${{ secrets.GH_ACTION_PUBLISHER_MFA_DEVICE_ARN }} \
-          --mfa-code ${{ inputs.mfaCode }}
-      - name: Add mask
-        run: |
-          echo "::add-mask::${{ env.AWS_ACCESS_KEY_ID }}" 
-          echo "::add-mask::${{ env.AWS_SECRET_ACCESS_KEY }}" 
-          echo "::add-mask::${{ env.AWS_SESSION_TOKEN }}" 
+          --mfa-code ${{ inputs.mfaCode }} \
+          --key ${{ secrets.GH_ACTION_PUBLISHER_AEM_KEY }}
       - uses: actions/checkout@v3
         with:
           repository: DataDog/datadog-agent
           ref: refs/heads/${{ inputs.agentBranch }}
           path: datadog-agent
       - name: Set up QEMU
-        uses: docker/setup-qemu-action@v1
+        uses: docker/setup-qemu-action@v2.1.0
       - run: |
          ./build-tools/bin/build_tools \
          build \
          --version 1 \
          --agent-version 1 \
          --architecture "${{ inputs.architecture }}" \
          --context-path "${GITHUB_WORKSPACE}" \
-         --destination-path "${GITHUB_WORKSPACE}/tmp"
-      - uses: actions/upload-artifact@v3
+         --destination-path "${GITHUB_WORKSPACE}/tmp" \
+         --docker-path "scripts_v2/Dockerfile.build" \
+         --artifact-name "datadog_extension.zip"
+      - name: Sign the layer
+        env:
+          AWS_ACCESS_KEY_ID: ${{ steps.auth.outputs.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ steps.auth.outputs.AWS_SECRET_ACCESS_KEY }}
+          AWS_SESSION_TOKEN: ${{ steps.auth.outputs.AWS_SESSION_TOKEN }}
+        run: |
+         ./build-tools/bin/build_tools \
+         sign \
+         --layer-path ./tmp/datadog_extension.zip \
+         --destination-path ./tmp/datadog_extension_signed.zip \
+         --key ${{ secrets.GH_ACTION_PUBLISHER_AEM_KEY }}
+      - uses: actions/upload-artifact@v3.1.1
+        with:
+          name: datadog-extension
+          path: ./tmp/datadog_extension_signed.zip
+          retention-days: 5
+      - id: list_region
+        env:
```

Comment:
nitpick: I think the `list_region` can be skipped with `if: ${{ github.event.inputs.saEast1Only == 'false' }}`?